### PR TITLE
launch token pool

### PR DESCRIPTION
**Pull request by PengDeng-Cyber**

### CODE REVIEW

File: src/periphery/LaunchBridge.sol [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import { UUPSUpgradeable } from "openzeppelin-contracts/contracts/proxy/utils/UUPSUpgradeable.sol";
+import { Ownable } from "openzeppelin-contracts/contracts/access/Ownable.sol";
+import { Pausable } from "openzeppelin-contracts/contracts/security/Pausable.sol";
+import { ReentrancyGuard } from "openzeppelin-contracts/contracts/security/ReentrancyGuard.sol";
+
+/**
+ * @title LaunchBridge
+ * @author CyberConnect
+ */
+contract LaunchBridge is
```
Comment (ZhimaoL) [inline src/periphery/LaunchBridge.sol:17]: It has nothing to do with bridge, change the name

File: src/periphery/LaunchBridge.sol [hunk 2]
```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity 0.8.14;
+
+import { IERC20 } from "openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+import { Initializable } from "openzeppelin-contracts/contracts/proxy/utils/Initializable.sol";
```
Comment (ZhimaoL) [inline src/periphery/LaunchBridge.sol:7]: Let's make this contract non-upgradable.

File: src/periphery/LaunchBridge.sol [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        deposits[to] += amount;
+        totalDeposits += amount;
+        emit Deposit(to, amount);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            ONLY OWNER 
+    //////////////////////////////////////////////////////////////*/
+
+    function withdraw(address to, uint256 amount) external onlyOwner {
```
Comment (ZhimaoL) [inline src/periphery/LaunchBridge.sol:73]: Should we allow for the withdrawal of noncyber tokens? e,g, native ETH or other ERC-20 was sent to this contract accidentally.

**Review by ZhimaoL — CHANGES_REQUESTED** [review]

File: src/periphery/LaunchBridge.sol (hunk 1 above)
Comment (ZhimaoL) [reply src/periphery/LaunchBridge.sol:17]: e.g.
TokenPool

File: src/periphery/LaunchBridge.sol (hunk 3 above)
Comment (PengDeng-Cyber) [reply src/periphery/LaunchBridge.sol:73]: transferring native token a contract with no `receive()` or `fallback() payable` defined will revert on evm.
And we can add erc-20 functions

**Review by ZhimaoL — APPROVED** [review]

**Pull request merged**